src/js/tracing/nativeframes.ts

```diff
@@ -180,25 +204,32 @@ export class NativeFramesInstrumentation {
       finishFrames = await NATIVE.fetchNativeFrames();
     }
 
-    _finishFrames.set(transaction.traceId, {
+    this._finishFrames.set(traceId, {
       nativeFrames: finishFrames,
       timestamp,
     });
 
-    _framesListeners.get(transaction.traceId)?.();
+    this._framesListeners.get(traceId)?.();
 
-    setTimeout(() => this._cancelFinishFrames(transaction), 2000);
+    setTimeout(() => this._cancelFinishFrames(span), 2000);
   }
 
   /**
    * On a finish frames failure, we cancel the await.
    */
-  private _cancelFinishFrames(transaction: Transaction): void {
-    if (_finishFrames.has(transaction.traceId)) {
-      _finishFrames.delete(transaction.traceId);
+  private _cancelFinishFrames(span: Span): void {
+    const traceId = spanToJSON(span).trace_id;
+    if (!traceId) {
+      return;
+    }
+
+    if (this._finishFrames.has(traceId)) {
+      this._finishFrames.delete(traceId);
 
       logger.log(
-        `[NativeFrames] Native frames timed out for ${transaction.op} transaction ${transaction.name}. Not adding native frames measurements.`,
+        `[NativeFrames] Native frames timed out for ${spanToJSON(span).op} transaction ${
+          spanToJSON(span).description
```

Comment:
NIT: Store the result of spanToJSON(span) to avoid redundant calls when accessing trace_id, op, and description.